### ISSUE TITLE
:bd on vimagit error

### ISSUE THREAD

**Issue by abaldwin88** [jreybert/vimagit#130]
I commonly use :bd to close vimagit (hard to overcome that muscle memory and use Q)

This has been working fine for months when I was using commit cafba10.  Recently I upgraded to commit 5921ef4 and now I am getting an error.  Here are my steps:

Modify a file --> Open Magit with `:Magit` --> Stage file --> close with `:bd` --> Open again with `:Magit` --> error

(This is happening in a brand new repo.  Just started it with `git init .`)

Note:  I do not see this error if I close with Q.

```
Error detected while processing BufEnter Auto commands for "magit:///Users/abaldwin/blarg/":
E121: Undefined variable: b:magit_current_commit_mode
```

OSX 10.11.6
iTerm 3.0.15
nvim 0.1.7
vimagit 5921ef4

P.S.

Awesome plugin by the way 👍

**Comment by jreybert** [comment]
Thanks for such a wonderful bug report!

You screw up all my buffer scoped variables with your `:bd` :)

**Comment by guyzmo** [comment]
just for the note: I'm getting the same error. Actually, I'm not *exactly* using `:bd` but [`:Bdelete`](https://github.com/moll/vim-bbye) which is like `:bd` as it closes the buffer, but *not* the current window.

(I'm actually only using `:MagitOnly`, and I'd like to either get rid of `q` or at least make it close the `Magit` buffer not the current split window)

**Comment by jreybert** [comment]
Hu @guyzmo , could you please open a new issue for your MagitOnly issue please?

EDIT: actually, thi issue already exists here: #132 

EDIT2: #132 was too broad, now in #137 Could you please give it a try?

**Comment by abaldwin88** [comment]
Just pulled down the fix.  Works like a charm 👍

**Comment by jreybert** [comment]
Merged in master and present in release 1.7.2
